src/main/java/zd/zero/waifu/motivator/plugin/player/WaifuSoundPlayerFactory.java

```diff
@@ -1,54 +1,58 @@
-package zd.zero.waifu.motivator.plugin.player;
-
-import org.apache.commons.lang3.StringUtils;
-
-import java.util.Optional;
-import java.util.stream.Stream;
-
-public final class WaifuSoundPlayerFactory {
-
-    private WaifuSoundPlayerFactory() {
-        throw new AssertionError( "Never instantiate." );
-    }
-
-    public static WaifuSoundPlayer createPlayer( String fileName ) {
-        String extension = Optional.ofNullable( fileName ).filter( f -> f.contains( "." ) )
-                .map( f -> f.substring( fileName.lastIndexOf( '.' ) + 1 ) ).orElse( "" );
-
-        Optional<SupportedFile> supportedFile = SupportedFile.ofExtension( extension );
-        if ( supportedFile.isPresent() ) {
-            switch ( supportedFile.get() ) {
-                case CLIP:
-                    return DefaultWaifuSoundPlayer.ofFile( fileName );
-
-                case MP3:
-                    return Mp3WaifuSoundPlayer.ofFile( fileName );
-            }
-        }
-
-        return EmptyWaifuSoundPlayer.of( fileName );
-    }
-
-    private enum SupportedFile {
-
-        CLIP( "wav", "au" ),
-
-        MP3( "mp3" );
-
-        private final String[] extensions;
-
-        SupportedFile( String... extensions ) {
-            this.extensions = extensions;
-        }
-
-        private static Optional<SupportedFile> ofExtension( String ext ) {
-            if ( StringUtils.isBlank( ext ) ) return Optional.empty();
-
-            return Stream.of( SupportedFile.values() )
-                    .filter( f -> Stream.of( f.extensions ).anyMatch( x -> x.equalsIgnoreCase( ext ) ) )
-                    .findFirst();
-        }
-
-    }
-
-}
+package zd.zero.waifu.motivator.plugin.player;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.nio.file.Path;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+public final class WaifuSoundPlayerFactory {
+
+    private WaifuSoundPlayerFactory() {
+        throw new AssertionError( "Never instantiate." );
+    }
+
+    public static WaifuSoundPlayer createPlayer( Path fileName ) {
+        String extension = Optional.ofNullable( fileName )
+            .map( Path::getFileName )
+            .map( String::valueOf )
+            .filter( f -> f.contains( "." ) )
+            .map( f -> f.substring( f.lastIndexOf( '.' ) + 1 ) ).orElse( "" );
```

Comment:
question (_not related_):  Do you have any idea why #84 happened? Because this block of code can be changed to 
```
String extension = FilenameUtils.getExtension( fileName );
```

I'm having `NoClassDefFoundError` on other platforms.

Reply:
![image](https://user-images.githubusercontent.com/15972415/91236273-2039e180-e6fd-11ea-8bc1-c4fed04205de.png)

It looks like it came bundled only with the IntelliJ platform, which would by why webstorm did the big sad.

However, that dependency is bundled with the plugin now https://github.com/zd-zero/waifu-motivator-plugin/blame/master/build.gradle#L22 so it can be changed.

Reply:
Do you want to make an issue?